python/ray/data/_internal/stats.py

```diff
@@ -439,6 +449,16 @@ def add(self, stats: DatasetStats) -> None:
             self.history_buffer.pop(0)
         self.count += 1
 
+    def add_pipeline_stats(self, other_stats: "DatasetPipelineStats"):
```

Comment:
I think the stats of other pipelines are not simply additive. The use case behind this change is adding a window (the first dataset), so can we just use existing `add()`?

Reply:
That will only add it to the history buffer. But we also need to add the iter stats and wait_time_s.

Reply:
Ok. Can we comment that `self` and `other_stats` need to cover disjoint windows of same pipeline?

Reply:
sounds good.